api/oidc.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrAudienceTooLong = errors.New("the API only supports at most one element in the audience")
+
+type OidcToken struct {
+	Token string `json:"token"`
+}
+
+func (c *Client) OidcToken(jobId string, audience ...string) (*OidcToken, *Response, error) {
```

Comment:
Does this `audience ...string` aspect of the method signature scale to handling future user-specified args (e.g. expiry/lifetime/TTL)?

I wonder if it should receive these parameters as a struct?
I wonder if `type oidcTokenRequest struct` should be public and received as an arg?
Not sure 🤷‍♂️

Reply:
Yeah, that's a good idea. I will do that.